yb-voyager/src/migassessment/sizing.go

```diff
@@ -43,22 +41,83 @@ type SourceDBMetadata struct {
 	Size            float64        `db:"size_in_bytes,string"`
 }
 
-var baseDownloadPath = "src/migassessment/resources/remote/"
+type ExpDataColocatedLimit struct {
+	maxColocatedSizeSupported  sql.NullFloat64 `db:"max_colocated_db_size_gb,string"`
+	numCores                   sql.NullFloat64 `db:"num_cores,string"`
+	memPerCore                 sql.NullFloat64 `db:"mem_per_core,string"`
+	maxSupportedNumTables      sql.NullInt64   `db:"max_num_tables,string"`
+	minSupportedNumTables      sql.NullFloat64 `db:"min_num_tables,string"`
+	maxSupportedSelectsPerCore sql.NullFloat64 `db:"max_selects_per_core,string"`
+	maxSupportedInsertsPerCore sql.NullFloat64 `db:"max_inserts_per_core,string"`
+}
+
 var DB *sql.DB
 var SourceMetaDB *sql.DB
+var fileName = "/yb_2024_0_source.db"
+
+//go:embed resources/yb_2024_0_source.db
+var experimentData20240 []byte
 
-func SizingAssessment() error {
+func SizingAssessment(assessmentMetadataDir string) error {
 	log.Infof("loading metadata files for sharding assessment")
 	sourceTableMetadata, sourceIndexMetadata, totalSourceDBSize := loadSourceMetadata()
-	createConnectionToExperimentData(TargetYBVersion)
-	shardedObjects, shardedObjectsSize :=
+
+	createConnectionToExperimentData(assessmentMetadataDir)
+	colocatedObjects, colocatedObjectsSize, coresToUse, shardedObjects :=
 		generateShardingRecommendations(sourceTableMetadata, sourceIndexMetadata, totalSourceDBSize)
 
 	// only use the remaining sharded objects and its size for further recommendation processing
-	generateSizingRecommendations(shardedObjects, shardedObjectsSize)
+	numNodes, optimalSelectConnections, optimalInsertConnections :=
+		generateSizingRecommendations(shardedObjects, totalSourceDBSize-colocatedObjectsSize, coresToUse)
+
+	// calculate time taken for colocated migration
+	migrationTimeForColocatedObjects, parallelThreadsColocated :=
+		calculateTimeTakenAndParallelThreadsForMigration("colocated", colocatedObjects,
```

Comment:
lets define const variable for this at the top.
const (
    COLOCATED = "colocated"
    SHARDED = "sharded"
)